organizations/admin/classes/domain/OrgEx.php

```diff
@@ -0,0 +1,333 @@
+    $query = $this->searchQuery($whereAdd, $orderBy, $limit, false);
+
+    $result = $this->db->processQuery($query, 'assoc');
+
+    $searchArray = array();
+
+    //need to do this since it could be that there's only one result and this is how the dbservice returns result
+    if (isset($result['OrganizationID'])) { $result = [$result]; }
+    foreach ($result as $row) {
+      $row = static::addIdsToOrganizationsRow($row);
+      array_push($searchArray, $row);
+    }
+    return $searchArray;
+  }
+
+
+
+  private static function addIdsToOrganizationsRow($row) {
+    $Organization = new OrgEx(new NamedArguments(array('primaryKey' => $row['OrganizationID'])));
+    $isbnOrIssns = $Organization->getIsbnOrIssn();
+    $row['isbnOrIssns'] = [];
+    foreach ($isbnOrIssns as $isbnOrIssn) {
+      array_push($row['isbnOrIssns'], $isbnOrIssn->isbnOrIssn);
+    }
+    return $row;
+  }
+
+
+
+  public function searchCount($whereAdd) {
+    $query = $this->searchQuery($whereAdd, '', '', true);
+    $result = $this->db->processQuery($query, 'assoc');
+
+    return $result['count'];
+  }
+
+
+
+  //returns array based on search for excel output (export.php)
+  public function export($whereAdd, $orderBy) {
+    $distinct_Organization_id_query = "SELECT DISTINCT(organizationID) AS Organization_id FROM Organization;";
+    $distinct_Organization_ids_assoc_array = $this->db->processQuery($distinct_Organization_id_query, 'assoc');
+    $distinct_Organization_ids = array_map(function($value) {
+      return $value["Organization_id"];
+    }, $distinct_Organization_ids_assoc_array);
+
+    $config = new Configuration();
+    if ($config->settings->organizationsModule == 'Y') {
+      $dbName = $config->settings->organizationsDatabaseName;
+      $orgJoinAdd = "LEFT JOIN $dbName.Organization O ON O.organizationID = R.organizationID";
+    }else{
+      $orgJoinAdd = "  LEFT JOIN Organization O ON O.organizationID = R.organizationID";
+    }
+
+    $whereStatement = "WHERE " . implode(" AND ", $whereAdd);
+
+    //now actually execute query
+    if(strlen($whereStatement) < 7){
+      $whereStatement = "";
+    }
+    $query = "
+SELECT
+  O.organizationID ,
+  O.name,
+  O.createDate,
+  O.updateDate,
+
+  A.name aName,
+
+  PO.name pName,
+
+  OrR.shortName,
+
+  C.name cName,
+  C.title,
+  C.addressText,
+  C.phoneNumber,
+  C.emailAddress
+
+
+
+FROM Organization O
+  LEFT JOIN OrganizationHierarchy OH ON OH.organizationID = O.organizationID
+  LEFT JOIN Organization PO ON PO.organizationID = OH.parentOrganizationID
+  LEFT JOIN OrganizationRoleProfile ORP ON ORP.organizationID = O.organizationID
+  LEFT JOIN OrganizationRole OrR ON OrR.organizationRoleID = ORP.organizationRoleID
+  LEFT JOIN Contact C ON C.organizationID = O.organizationID
+  LEFT JOIN Alias A ON A.organizationID = O.organizationID
+  LEFT JOIN ContactRoleProfile CRP ON CRP.contactID = C.contactID
+  LEFT JOIN ContactRole CR ON CR.contactRoleID = CRP.contactRoleID 
```

Comment:
All of these joins mean that an organization export may have more rows in the spreadsheet than hits indicated by the organization search. For example, an organization with aliases will show up multiple times in the export.

That may well be OK, as consumers of the export may prefer that to splitting apart repeats merged together via GROUP_CONCAT(), but it's probably worth a note in the documentation.